src/apps/somm/positions.ts

```diff
@@ -46,26 +48,43 @@ const hook: PositionsHook = {
           }
         }
 
-        const [underlyingAsset, underlyingAssetSymbol, cellarName] =
-          await Promise.all([
-            client.readContract({
-              address: cellar.address,
-              abi: cellarV0821Abi,
-              functionName: 'asset',
-            }),
-            client.readContract({
-              address: cellar.address,
-              abi: cellarV0821Abi,
-              functionName: 'symbol',
-            }),
-            client.readContract({
-              address: cellar.address,
-              abi: cellarV0821Abi,
-              functionName: 'name',
-            }),
-          ])
+        const [
+          underlyingAsset,
+          underlyingAssetSymbol,
+          cellarName,
+          cellarDecimals,
```

Comment:
is this underlying asset decimals? or does the cellar have decimals as well?
sorry, i'm not well versed in how somm works, but it caught my eye

Reply:
yes the cellar has decimals too 😅 i think that usually the cellar token has the same decimals as the underlying token, but i'm not sure that it is a necessity. since we use the cellar value for withdrawals, i thought it'd be more safe to use the cellar decimals directly